### PR TITLE
Add microbenchmarks, improve RangeMap perf

### PR DESCRIPTION
**Pull request by rbuckton**
This adds microbenchmarks to analyze the performance of several parts of Deopt Explorer and improves the performance of `RangeMap`, which is used throughout the extension to collection information related to positions and ranges, such as for producing hovers.

The key changes were to support a user-defined comparer in `SplayTree`, which allowed `RangeMap` to switch from using series of nested `SplayTree`s that map `startLine -> startCharacter -> endLine -> endCharacter -> T` to a single `SplayTree` whose keys are `Range`s using a custom comparer. The ranges are sorted by `range.start` in ascending order, then `range.end` in descending order. This order allows us to treat subsequent ranges as if they were contained by prior ranges as long as they share the same start position, which results in more efficient traversal.

This also adds linked list-like pointers to `SplayTree.Node` to make in-order traversal more efficient and reliable.

The net result is a 40x improvement in calls to `.findAllContaining()`:

image

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -30,7 +31,7 @@
 			"outFiles": ["${workspaceFolder}/out/**/*.js"],
 			"smartStep": true,
 			"autoAttachChildProcesses": true,
-			"skipFiles": ["<node_internals>/**","${workspaceFolder}/node_modules/tslib/**"],
+			"skipFiles": ["<node_internals>/**","${workspaceFolder}/node_modules/tslib/**", "C:/users/rbuckton/AppData/Local/Programs/**"],
```
Comment (jakebailey) [inline .vscode/launch.json:34]: Is this leftover?

Comment (rbuckton) [reply .vscode/launch.json:34]: Ah, yeah. Debugging and stepping over vscode internals since they weren't excluded.

**Comment by jakebailey** [comment]
Long-term, I wonder if you'd have it easier running `vitest` and its built-in benchmarking; I've found it to be pretty handy and avoids a lot of boilerplate.

**Comment by rbuckton** [comment]
At this point I'm pretty familiar with `jest`'s quirks. I'm not sure I want to switch to a different test framework just for benchmarking.

My biggest issue at the moment is bit rot with my CI builds..

**Review by jakebailey — APPROVED** [review]

**Pull request merged**